### DISCUSSION TITLE
Problem with BEP+BEM (single_point model)

### DISCUSSION THREAD

**Discussion by KevinInSomeWhere** [NCAR/hrldas#170] [Q&A]
When I set the ZLVL 50 as well as URBAN_ATMOSPHERE_THICKNESS = 5.0 in the hrldas namelist，the model only worked for 23 hours, but the same error appered again. I checked my namelist, to make sure that ZLVL has a numerical relationship with URBAN_ATMOSPHERE_THICKNESS.
![Q](https://github.com/NCAR/hrldas/assets/107543049/cf87a925-cb1e-4b3f-908d-4fe0f77075af)
In addtion, I checked the hrldas.log, I found out there is the '**-999**' value existed. 
How do I adjust the parameters to keep the model running?
![Q2](https://github.com/NCAR/hrldas/assets/107543049/b9a07b8d-4318-431b-9a43-33a67b0fd7d1)

**Comment by cenlinhe** [comment]
another follow up question from the same user:
" 
How do I adjust the parameters to keep the model running?
    I turn on the green roof scenario for URBPARAM.TBL(BEM+BEP) inside the time period that Single_point model can run, but I noticed that T2M(Case-Base) didn't change. However, SLUCM, which you suggested to        me earlier, was able to detect the temperature difference. 
     How can green roof  be implemented in BEM+BEP?
"

**Reply by cenlinhe** [reply]
1. To answer your T2M question: by default, the T2M in HRLDAS/Noah-MP is only for non-urban (i.e., rural) portion of a grid, that is why you did not see any changes. You will need to add a diagnostic T2M variable from the BEP+BEM and output it. We realize this needs to be improve and will do this later this year.
2. To answer your green roof in BEM+BEP question: currently the green roof and solar panel roof are already in BEP+BEM.
3. For your model crash error, it seems due to the ZLVL and URBAN_ATMOSPHERE_THICKNESS set up. We do not have a good solution because these setups vary over different urban regions/environments. Could you please try the following: 
reduce the hard-coded dz_u value to 2.0 in ../urban/wrf/module_sf_bep_bem.F and change zlvl to 25 and change URBAN_ATMOSPHERE_THICKNESS to 2.

**Reply by KevinInSomeWhere** [reply]
Thank you for your response! I changed the values according to the values you provided, but I still get the same error, also I would like to ask if I need to add additional RH variables, according to the formula provided by WRF inside the formula I need PSFC, Q2,and T2, my question is that Q2 has to be re-added to the city grid points for the calculations as well?

**Reply by cenlinhe** [reply]
Sorry, for the model crash error, I do not have a good solution currently. I am going to have a discussion with Prof. Alberto Martilli (original developer for BEP-BEM) to see if there is any solution.
For Q2, there is a variable "qv_u" in BEP-BEM code for the specific humidity over urban for each vertical level, maybe you could use this to infer Q2.
